### DISCUSSION TITLE
Release notes for Google Play

### DISCUSSION THREAD

**Discussion by maurovanetti** [codemagic-ci-cd/codemagic-docs#1050] [Q&A] [answered]
Hi! This is my first question here.

We publish our Flutter app simultaneously to Google Play and App Store (and Huawei AppGallery, actually). This app is in Italian only.
We have put the release notes in a text file called `release_notes_it.txt`, which worked perfectly for iOS but failed with Google Play. The workflow completed and the new release was published but it has empty release notes.
What's wrong? Thanks for your help.

Documentation: https://docs.codemagic.io/yaml-notification/publish-release-notes/#setting-up-release-notes

**Comment by himesh-cm** [comment, marked as answer]
Do you mean the other way around?
Because apple supports it `.json` only.

**Reply by maurovanetti** [reply]
No, it did work for App Store. The documentation is a bit confusing, but if you check the section called _One file per language used_, it mentions files in the `release_notes_<lang>.txt` format.

I didn't follow the instructions in the section called _Single file_ because I don't want it to default to `en-US` for Google Play.

**Reply by himesh-cm** [reply]
Any error on the build page?

**Reply by maurovanetti** [reply]
No errors, the log is very short:
```
Publishing app-release.aab to Google Play
Published app-release.aab to track production
```

**Reply by maurovanetti** [reply]
The iOS logs are very verbose instead. This is the relevant part (I've removed some data):
```
Update App Store Version Localization for locale it
-- App Store Version Localization --
Id: …
Type: appStoreVersionLocalizations
Description: Questa app permette l'accesso ai servizi di mensa dell'ente regionale per il diritto allo studio e alla conoscenza … Può essere utilizzata solo dagli utenti già registrati sul sito …
Keywords: …
Locale: it
Support url: …
Whats new: Nuova grafica e rinnovo generale dell'app.
Schermo sempre acceso a massima luminosità durante la visualizzazione del tesserino virtuale.
Migliorie tecniche nella gestione degli accessi.
```

**Reply by himesh-cm** [reply]
@maurovanetti, can you please check if the default language in your store listing in the google play console is set to Italian?

**Reply by maurovanetti** [reply]
Yes, the default language is Italian. It's the only language actually.
(Thank you again for your help.)

**Reply by himesh-cm** [reply]
In that case, can you try the below solution if it works?
Just keep `release_notes.txt `instead of `release_notes_it.txt` 

This will fail for iOS, but it will help debug the issue, so maybe you want it to test in the internal track or something.

Make sure it's in the root directory. Let me know how it goes.

**Reply by maurovanetti** [reply]
For the way the workflow is organised it's a bit complicated to launch a test run that includes the deploy step (we have the internal test tracks on DeployGate instead of the official stores), but we have many releases and I can easily do an experiment soon. I might even automatically rename the file for Android only.

**Reply by maurovanetti** [reply]
OK, I could not test `release_notes.txt` but I found something else that may lead us to the explanation of this issue.

I have now managed to have a completely automated system by creating a `release_notes.json` file like this:
```
[
  {
    "language": "it-IT",
    "text": "..."
  },
  {
    "language": "it",
    "text": "..."
  }
]
```
Apparently, for the Italian locale App Store uses `it` and Google Play uses `it-IT`. I guess this also explains why `release_notes_it.txt` worked for Apple but failed for Google: the latter expected `release_notes_it-IT.txt`.

**Reply by himesh-cm** [reply]
Thanks for sharing @maurovanetti
